### PR TITLE
Kangaskhan boss

### PR DESCRIPTION
**Pull request by wiz1989**
Here is all the things that went into this:

-     new trainer struct TrainerMonItemAbilityCustomMoves to also customize trainer mon abilities
-     added lockedAbility functionality from ghoulslash to give any mon any ability using SetMonLockedAbility()
-     Math Club item that doubles the amount of strikes of any attacking move for Cubone/Marowak (so with this item Bonemerang hits four times in total)
-     Gen V Sturdy mechanic
-     new berry Liberry that works like a max potion
-     new substitute move (sub teacher) with kanga baby sprite as doll replacement and additional effects:
  -     healing to 100% instead of dealing 1/4th damage
  -     recycling a used item

### CODE REVIEW

File: include/battle.h [hunk 1]
```diff
@@ -223,6 +235,8 @@ struct SpecialStatus
     u8 ppNotAffectedByPressure:1;
     u8 faintedHasReplacement:1;
     u8 focusBanded:1;
+    u8 focusSashed:1;
```
Comment (ravepossum) [inline include/battle.h]: Looks like potentially a duplicate change with PR #22 ? Same comment applies - I think we need fix the alignment here, maybe slap these both as bitfields of one of the unused fields.

File: include/battle.h [hunk 2]
```diff
@@ -725,3 +739,5 @@ extern struct MultiBattlePokemonTx gMultiPartnerParty[3];
 extern u16 gRandomTurnNumber;
 
 #endif // GUARD_BATTLE_H
+
+#define BATTLER_MAX_HP(battlerId)(gBattleMons[battlerId].hp == gBattleMons[battlerId].maxHP)
```
Comment (ravepossum) [inline include/battle.h]: Looks like another duplicate change with PR #22 ?

File: include/pokemon.h [hunk 3]
```diff
@@ -54,7 +55,7 @@ struct PokemonSubstruct3
  /* 0x05 */ u32 spAttackIV:5;
  /* 0x06 */ u32 spDefenseIV:5;
  /* 0x07 */ u32 isEgg:1;
- /* 0x07 */ u32 abilityNum:1;
+ /* 0x07 */ u32 unk7:1;
```
Comment (ravepossum) [inline include/pokemon.h:58]: If this is now filler, I'd probably name it `filler` instead

File: include/pokemon.h [hunk 4]
```diff
@@ -183,7 +185,7 @@ struct BattlePokemon
     /*0x16*/ u32 spAttackIV:5;
     /*0x17*/ u32 spDefenseIV:5;
     /*0x17*/ u32 isEgg:1;
-    /*0x17*/ u32 abilityNum:1;
+    /*0x17*/ u32 abilityNum:2;
```
Comment (ravepossum) [inline include/pokemon.h:188]: Maybe ghoul knows something I don't but this also seems like it doesn't align as intended. Maybe slap these at the end or cannibalize the `unknown` field that seems unused?

File: src/data/text/move_names.h [hunk 5]
```diff
@@ -354,5 +354,6 @@ const u8 gMoveNames[MOVES_COUNT][MOVE_NAME_LENGTH + 1] = {
     [MOVE_WATER_PULSE]   = _("WATER PULSE"),
     [MOVE_DOOM_DESIRE]   = _("DOOM DESIRE"),
     [MOVE_PSYCHO_BOOST]  = _("PSYCHO BOOST"),
-    [MOVE_HEART_SWAP]    = _("HEART SWAP")
+    [MOVE_HEART_SWAP]    = _("HEART SWAP"),
+    [MOVE_SUBSTITUTE_TEACHER]    = _("SUBSTITUTE")
```
Comment (ravepossum) [inline src/data/text/move_names.h:359]: Is Substitute Teacher intended to just be named Substitute?

File: src/pokemon.c [hunk 6]
```diff
@@ -3776,12 +3782,21 @@ u8 GetMonsStateToDoubles(void)
...
-        gLastUsedAbility = gSpeciesInfo[species].abilities[1];
-    else
-        gLastUsedAbility = gSpeciesInfo[species].abilities[0];
+    switch (abilityNum)
+    {
+    case 0:
+    case 1:
+        gLastUsedAbility = gSpeciesInfo[species].abilities[abilityNum];
+        break;
+    case 2: //hidden ability
```
Comment (ravepossum) [inline src/pokemon.c]: Should we set `gLastUsedAbility` to a default value in case 2 and at the end to avoid potential unexpected behavior?

**Review by ravepossum — COMMENTED** [review]
No show-stoppers here either, but a few things that I think are worth a look - especially making sure the duplicates get ironed out.

File: include/battle.h (hunk 1 above)
Comment (wiz1989) [reply include/battle.h]: correct, had to duplicate some stuff for sturdy and tried to make the change 1:1 with #22 so it is easier to merge.

File: include/battle.h (hunk 2 above)
Comment (wiz1989) [reply include/battle.h]: correct, had to duplicate some stuff for sturdy and tried to make the change 1:1 with #22 so it is easier to merge.

File: include/pokemon.h (hunk 3 above)
Comment (wiz1989) [reply include/pokemon.h:58]: none of these bytes are actually correctly filled. It looks odd, but I will keep it as it is successfully tested.

File: src/data/text/move_names.h (hunk 5 above)
Comment (wiz1989) [reply src/data/text/move_names.h:359]: Yes. Just Celia things :D

File: include/pokemon.h (hunk 4 above)
Comment (wiz1989) [reply include/pokemon.h:188]: same thing as above. Won't touch these bits. Even though this one here at least makes sense (5+1+2=8)

File: src/pokemon.c (hunk 6 above)
Comment (wiz1989) [reply src/pokemon.c]: Hidden abilities don't exist in FRLG. So simply deleted case 2.

**Review by ravepossum — APPROVED** [review]

**Pull request merged**